### PR TITLE
AVT-937, AVT-938, AVT-939: Minimize on close

### PR DESCRIPTION
**Pull request by tranducanh**
### This PR is:
- [x] new test
- [ ] test stabilization
- [ ] framework improvement
- [ ] bug fix

### Description
Verify Minimize on Close option once the application is installed

### Run
[Report_Result.zip](https://github.com/symphonyoss/SymphonyElectron/files/2187846/Report_Result.zip)

### CODE REVIEW

File: tests/spectron/spectronConstants.js [hunk 1]
```diff
@@ -7,4 +7,9 @@ module.exports = {
 
     SEARCH_LIBRARY_PATH_MAC: "node_modules/electron/dist/Electron.app/Contents/library",
     SEARCH_LIBRARY_PATH_WIN: "node_modules/electron/dist/library",
+    TESTED_HOST: "https://cip4-qa.symphony.com/",
+    ADMIN_USERNAME: "spectron_auto_admin",
```
Comment (thaisym1912) [inline tests/spectron/spectronConstants.js]: Remove User/Pass as redundant

File: tests/spectron/spectronWindowsActions.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+            robot.setMouseDelay(500);
+            robot.moveMouse(bounds.x + 200, bounds.y + 15);
+            robot.mouseClick("left", "double");
+        })
+        await this.app.browserWindow.isMaximized().then(function (maximized) {
+            expect(maximized).toBeTruthy();
+        })
+    }
+
+    async relaunchApp(configurations){
```
Comment (thaisym1912) [inline tests/spectron/spectronWindowsActions.js]: Please consider to use  app.startApplication() instead.

File: tests/spectron/spectronWindowsActions.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+                console.log("isMinimizedWindows: Quit");
+                return rminimized;
+            });
+          
+       return rminimized;
+    }
+    async minimizeWindowByClick()
+    {
+        console.log("minimizeWindowByClick");
+        await this.app.client.click("button#title-bar-minimize-button");
```
Comment (thaisym1912) [inline tests/spectron/spectronWindowsActions.js]: Please create a file to store locators.

File: tests/spectron/spectronWindowsActions.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+                rminimized = constants.QUIT;
+                console.log("isMinimizedWindows: Quit");
+                return rminimized;
+            });
+          
+       return rminimized;
+    }
+    async minimizeWindowByClick()
+    {
+        console.log("minimizeWindowByClick");
```
Comment (thaisym1912) [inline tests/spectron/spectronWindowsActions.js]: Remove console log

File: tests/spectron/spectronWindowsActions.js [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+        await this.app.client.click("button#title-bar-minimize-button"); 
+    }
+    async closeWindowByClick()
+    {
+        console.log("closeWindowByClick");
```
Comment (thaisym1912) [inline tests/spectron/spectronWindowsActions.js]: Remove console log

File: tests/spectron/spectronWindowsActions.js [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+        await this.app.client.click("button#title-bar-close-button"); 
+    }
+    async openApplicationMenuByClick()
+    {
+        console.log("openApplicationMenuByClick");
```
Comment (thaisym1912) [inline tests/spectron/spectronWindowsActions.js]: Remove console log

File: tests/spectron/deselect-minimize-on-close.spectron.js [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+                    });
+                } else {
+                    done();
+                }
+            })
+        } else {
+            done();
+        }
+    });
+    it('should bring the app to top', () => {
```
Comment (thaisym1912) [inline tests/spectron/deselect-minimize-on-close.spectron.js]: Consider to use startApplication in this PR: https://github.com/symphonyoss/SymphonyElectron/pull/430/files

File: tests/spectron/deselect-minimize-on-close.spectron.js [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+        app.browserWindow.focus();
+        return app.browserWindow.setAlwaysOnTop(true).then(() => {
+            return app.browserWindow.isAlwaysOnTop().then((isOnTop) => {
+                expect(isOnTop).toBeTruthy();
+            });
+        });
+    });
+    /**
+     * Keep size and position of the windows in previous session
```
Comment (thaisym1912) [inline tests/spectron/deselect-minimize-on-close.spectron.js]: Update description of this test

File: tests/spectron/deselect-minimize-on-close.spectron.js [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+                if (userConfig.minimizeOnClose == false) {
+                    await wActions.selectMinimizeOnClose();
+                    await wActions.selectMinimizeOnClose();
+                    await wActions.closeWindowByClick();
+                    await wActions.verifyMinimizeWindows();
+                }
+                //When app  ticked on Minimize On Close Menu Item
+                //Select 1 times to perform for un-tick 
+                else {
+                    console.log("userConfig.minimizeOnClose::true::" + userConfig.minimizeOnClose);
```
Comment (thaisym1912) [inline tests/spectron/deselect-minimize-on-close.spectron.js]: Remove console log

File: tests/spectron/deselect-minimize-on-close.spectron.js [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+                    done();
+                }
+            })
+        } else {
+            done();
+        }
+    });
+   
+    /**
+     * Keep size and position of the windows in previous session
```
Comment (thaisym1912) [inline tests/spectron/deselect-minimize-on-close.spectron.js]: same here

File: tests/spectron/minimize-on-close-icon.spectron.js [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+                    done();
+                }
+            })
+        } else {
+            done();
+        }
+    });
+  
+    /**
+    * Keep size and position of the windows in previous session
```
Comment (thaisym1912) [inline tests/spectron/minimize-on-close-icon.spectron.js]: Same here

File: tests/spectron/spectronConstants.js (hunk 1 above)
Comment (tranducanh) [reply tests/spectron/spectronConstants.js]: fixed

File: tests/spectron/deselect-minimize-on-close.spectron.js (hunk 9 above)
Comment (tranducanh) [reply tests/spectron/deselect-minimize-on-close.spectron.js]: fix

File: tests/spectron/minimize-on-close-icon.spectron.js (hunk 11 above)
Comment (tranducanh) [reply tests/spectron/minimize-on-close-icon.spectron.js]: fixed

File: tests/spectron/spectronWindowsActions.js (hunk 2 above)
Comment (tranducanh) [reply tests/spectron/spectronWindowsActions.js]: fixed

File: tests/spectron/spectronWindowsActions.js (hunk 3 above)
Comment (tranducanh) [reply tests/spectron/spectronWindowsActions.js]: fixed

File: tests/spectron/spectronWindowsActions.js (hunk 4 above)
Comment (tranducanh) [reply tests/spectron/spectronWindowsActions.js]: fixed

File: tests/spectron/spectronWindowsActions.js (hunk 5 above)
Comment (tranducanh) [reply tests/spectron/spectronWindowsActions.js]: fixed

File: tests/spectron/spectronWindowsActions.js (hunk 6 above)
Comment (tranducanh) [reply tests/spectron/spectronWindowsActions.js]: fixed

File: tests/spectron/deselect-minimize-on-close.spectron.js (hunk 7 above)
Comment (tranducanh) [reply tests/spectron/deselect-minimize-on-close.spectron.js]: fixed

File: tests/spectron/deselect-minimize-on-close.spectron.js (hunk 8 above)
Comment (tranducanh) [reply tests/spectron/deselect-minimize-on-close.spectron.js]: fixed

**Comment by VishwasShashidhar** [comment]
@maoo Can you please check why the CLA is failing here? It seems to be fine on the other PRs for the same committer.

**Comment by maoo** [comment]
@VishwasShashidhar , looking at the logs, I think the issue is https://github.com/ColinEberhardt/cla-bot/issues/74 (and more info on https://github.com/octokit/octokit.net/issues/1433#issuecomment-235714712) ; the cla-bot fails if the email address configured in the GitHub author's account. @tranducanh , could you please check?

**Pull request closed without merging**